scratchml/models/svc.py

```diff
@@ -0,0 +1,202 @@
+from scratchml.utils import split_data_into_batches
+from ..metrics import accuracy, recall, precision, f1_score, confusion_matrix
+from scratchml.regularizations import l2
+from scratchml.scalers import StandardScaler
+import numpy as np
+
+
+class SVC:
+    """
+    Support Vector Classifier (SVC) with options for linear, polynomial, and RBF kernels.
+    """
+
+    _valid_metrics = ["accuracy", "precision", "recall", "f1_score", "confusion_matrix"]
+
+    def __init__(
```

Comment:
Please, add typing for parameters and outputs for all functions. Here is an example of how to do it using that function:

```python3
def __init__(
   self,
   C: float = 0.35,
   alpha: float = 0.01,
   kernel: str = "linear",
   degree: int = 3,
   max_iter: int = 1500,
   tol: float = 1e-4,
   learning_rate: float = 1e-5,
   decay: float = 0.999,
   batch_size: int = 64,
   early_stopping: bool = True,
   adaptive_lr: bool = False,
) -> None:
```

Remember, for some types (like List, Tuple, and others), you need to import the built-in 'Typing' library. You can do that with the following code:

```python3
from typing import List, Tuple, Dict
```

Reply:
Please, add a docstring as well. Like this:

```python3
def init(...) -> None:
    """
    Creates a Support Vector Classifier (SVC) instance.

    Args:
       C (float): the parameter's description goes here. Defaults to 0.35.
       alpha (float): the parameter's description goes here. Defaults to 0.01.
       kernel (str): the parameter's description goes here. Put the valid parameters as well. Defaults to "linear".
       degree (int): the parameter's description goes here. Defaults to 3.
       max_iter (int): the parameter's description goes here. Defaults to 1500.
       tol (float): the parameter's description goes here. Defaults to 1e-4.
       learning_rate (float): the parameter's description goes here. Defaults to 1e-5.
       decay (float): the parameter's description goes here. Defaults to 0.999.
       batch_size (int): the parameter's description goes here. Defaults to 64.
       early_stopping (bool): the parameter's description goes here. Defaults to True.
       adaptive_lr (bool): the parameter's description goes here. Defaults to False.
    """
```